frontend/src/HomePage/AppCard.jsx

```diff
@@ -169,14 +211,25 @@ export default function AppCard({
           </div>
         </div>
         <div>
-          <ToolTip trigger={['hover']} message={app.name}>
+          {isNameOverflowing ? (
+            <ToolTip trigger={['hover']} message={app.name}>
+              <h3
+                ref={h3Ref}
+                className="app-card-name font-weight-500 tj-text-md"
+                data-cy={`${app.name.toLowerCase().replace(/\s+/g, '-')}-title`}
+              >
+                {decodeEntities(app.name)}
+              </h3>
+            </ToolTip>
+          ) : (
             <h3
+              ref={h3Ref}
               className="app-card-name font-weight-500 tj-text-md"
               data-cy={`${app.name.toLowerCase().replace(/\s+/g, '-')}-title`}
             >
               {decodeEntities(app.name)}
             </h3>
-          </ToolTip>
+          )}
```

Comment:
Can you create a function and reuse the repeated code?

Reply:
sure
looking into it